pkg/event_handler/main.go

```diff
@@ -27,14 +27,24 @@ func Start(ctx context.Context, stop context.CancelFunc, cfg *conf.GlobalConfig,
 		Notifier: notifier,
 	}
 	go func() {
-		for event := range watcher.ResultChan() {
-			err2 := handler.Handle(ctx, &event)
-			if err2 != nil {
-				log.Printf("[event handler] failed to Handle workflow event %s", err2) // ERROR
+		for {
+			select {
+			case <-ctx.Done():
+				log.Print("[event handler] context canceled, stopping watcher")
+				watcher.Stop()
+				return
+			case event, ok := <-watcher.ResultChan():
+				if !ok {
+					log.Print("[event handler] result channel closed")
+					watcher.Stop()
+					stop()
+					return
+				}
+				if err2 := handler.Handle(ctx, &event); err2 != nil {
+					log.Printf("[event handler] failed to Handle workflow event: %v", err2)
+				}
 			}
 		}
-		log.Print("[event handler] stopped work, closing watcher")
```

Comment:
@GoshaDo can explain how the race condition could happen ?

Reply:
If stop() is called, it will cancel the context (ctx). If the context is already canceled before watcher.Stop() completes, there could be race conditions or unexpected behavior.

Reply:
ok there are no race condition, but it is a good practice to stop the process if the watcher Chanel closed.